php-checks/src/main/java/org/sonar/php/checks/security/RegexUsageCheck.java

```diff
@@ -63,9 +74,37 @@
 
   @Override
   protected void createIssue(FunctionCallTree tree) {
-    if (!tree.arguments().isEmpty()) {
-      context().newIssue(this, tree, MESSAGE);
+    int index = getPatternArgumentIndex(tree);
+
+    if (tree.arguments().size() <= index) {
+      return;
+    }
+
+    ExpressionTree patternArgument = tree.arguments().get(index);
+    if (patternArgument.is(Kind.REGULAR_STRING_LITERAL)) {
+      String value = ((LiteralTree) patternArgument).value();
+      if (value.length() >= MIN_PATTERN_LENGTH && hasEnoughNumberOfSpecialChars(value)) {
+        context().newIssue(this, tree, MESSAGE);
+      }
+    }
+  }
+
+  private int getPatternArgumentIndex(FunctionCallTree tree) {
+    int index = 0;
+    if (tree.callee().toString().equalsIgnoreCase(MB_EREG_SEARCH_INIT)) {
+      index = 1;
+    }
+    return index;
+  }
+
+  private boolean hasEnoughNumberOfSpecialChars(String value) {
+    int numberOfSpecialChars = 0;
+    for (char c : value.toCharArray()) {
+      if (SPECIAL_CHARS.contains(c)) {
+        numberOfSpecialChars++;
+      }
     }
+    return numberOfSpecialChars >= 2;
```

Comment:
we should return from the loop once this var is greater than 2.
One way to do also get rid of the set could be to use a switch here : 

``` 
for () {
switch(c) {
  case '+': 
  case '*': 
  case '{': 
    nb++;
    break;
}
if(nb == 2) return true;
}
return false;
```